client/www/pages/status.tsx

```diff
@@ -0,0 +1,349 @@
+      <div className="font-mono flex-1 max-w-4xl">
+        <span className="text-sm md:text-base font-semibold">
+          Overall Uptime
+        </span>
+        <div className="flex shadow-sm bg-white border-gray-200 border-2 p-3 md:p-4 mt-2 h-48 md:h-64 justify-center items-center">
+          <div className="flex items-center flex-row gap-10">
+            <div className="flex flex-col w-32 text-center">
+              <span className="text-2xl font-semibold">
+                {uptimeData?.overall_uptime?.['24h']?.toFixed(3) || '100.000'}%
+              </span>
+              <span className="text-sm text-gray-500">Last 24 Hours</span>
+            </div>
+            <div className="w-0.5 h-20 bg-gray-200"></div>
+            <div className="flex flex-col w-32 text-center">
+              <span className="text-2xl font-semibold">
+                {uptimeData?.overall_uptime?.['7d']?.toFixed(3) || '99.910'}%
+              </span>
+              <span className="text-sm text-gray-500">Last 7 Days</span>
+            </div>
+            <div className="w-0.5 h-20 bg-gray-200"></div>
+            <div className="flex flex-col w-32 text-center">
+              <span className="text-2xl font-semibold">
+                {uptimeData?.overall_uptime?.['30d']?.toFixed(3) || '99.837'}%
+              </span>
+              <span className="text-sm text-gray-500">Last 30 Days</span>
+            </div>
+            <div className="w-0.5 h-20 bg-gray-200"></div>
+            <div className="flex flex-col w-32 text-center">
+              <span className="text-2xl font-semibold">
+                {uptimeData?.overall_uptime?.['90d']?.toFixed(3) || '99.621'}%
+              </span>
+              <span className="text-sm text-gray-500">Last 90 Days</span>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function StatusPage() {
+  const [uptimeData, setUptimeData] = useState<any>(null);
+  const [loading, setLoading] = useState(true);
+  const [lastUpdated, setLastUpdated] = useState(new Date());
+  const [nextUpdate, setNextUpdate] = useState(60);
+
+  const fetchUptimeData = async () => {
+    try {
+      const response = await fetch('/api/uptime');
+      const UptimeRobotResponse = await response.json();
+      const processedData = processUptimeData(UptimeRobotResponse);
+      setUptimeData(processedData);
+      setLastUpdated(new Date());
+      setNextUpdate(60);
+      setLoading(false);
+    } catch (error) {
+      console.error('Failed to fetch uptime data:', error);
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      setNextUpdate((prev) => {
+        if (prev <= 1) {
+          fetchUptimeData();
+          return 60;
+        }
+        return prev - 1;
+      });
+    }, 1000);
+
+    return () => clearInterval(timer);
+  }, []);
+
+  useEffect(() => {
+    fetchUptimeData();
```

Comment:
Nit: it may be better to run `fetchUptimeData` _once_ on the server first. This way when the page loads we have data.